app/controllers/guides_controller.rb

```diff
@@ -61,9 +61,15 @@ def confirm_unpublish
     )
 
     guide_manager = GuideManager.new(guide: guide, user: current_user)
-    guide_manager.unpublish(redirect)
+    result = guide_manager.unpublish(redirect)
 
-    redirect_to edit_guide_path(guide), notice: "Guide has been unpublished"
+    if result.success?
+      redirect_to edit_guide_path(guide), notice: "Guide has been unpublished"
+    else
+      redirect_to unpublish_guide_path(guide), flash: {
+        error: "Guide could not be unpublished"
+      }
```

Comment:
It's common to re-render the template for the "new" action when something fails. eg. https://github.com/alphagov/service-manual-publisher/blob/master/app/controllers/guides_controller.rb#L78

The benefit is that form fields get repopulated for free (most of the time). In this case we're probably not loosing a lot.. but ya know.. it's a pattern people expect I guess.